vector/src/main/java/im/vector/riotredesign/core/platform/StateView.kt

```diff
@@ -62,28 +62,30 @@ class StateView @JvmOverloads constructor(context: Context, attrs: AttributeSet?
 
     private fun update(newState: State) {
         when (newState) {
-            is StateView.State.Content -> {
+            is State.Content -> {
                 progressBar.visibility = View.INVISIBLE
                 errorView.visibility = View.INVISIBLE
                 emptyView.visibility = View.INVISIBLE
                 contentView?.visibility = View.VISIBLE
             }
-            is StateView.State.Loading -> {
+            is State.Loading -> {
                 progressBar.visibility = View.VISIBLE
                 errorView.visibility = View.INVISIBLE
                 emptyView.visibility = View.INVISIBLE
                 contentView?.visibility = View.INVISIBLE
             }
-            is StateView.State.Empty -> {
+            is State.Empty   -> {
                 progressBar.visibility = View.INVISIBLE
                 errorView.visibility = View.INVISIBLE
                 emptyView.visibility = View.VISIBLE
+                emptyImageView.setImageDrawable(newState.image)
                 emptyMessageView.text = newState.message
+                emptyTitleView.text = newState.title
                 if (contentView != null) {
                     contentView!!.visibility = View.INVISIBLE
```

Comment:
No need to to force optional unwrapping? Or just use contentView?.visibility (shorter cleaner?)

Reply:
Done